lib/core/utils/preload-cssom.js

```diff
@@ -1,58 +1,82 @@
 /**
- * Returns a then(able) queue of CSSStyleSheet(s)
- * @param {Object} ownerDocument document object to be inspected for stylesheets
- * @param {number} timeout on network request for stylesheet that need to be externally fetched
- * @param {Function} convertTextToStylesheetFn a utility function to generate a style sheet from text
- * @return {Object} queue
+ * Make an axios get request to fetch a given resource and resolve
+ * @method getExternalStylesheet
+ * @param {Object} options an object with properties to configure the external XHR
+ * @property {Object} options.resolve resolve callback on queue
+ * @property {Object} options.reject reject callback on queue
+ * @property {String} options.url string representing the url of the resource to load
+ * @property {Object} options.rootNode document or shadowDOM root document for which to process CSSOM
+ * @property {Number} options.timeout timeout to about network call
+ * @property {Function} options.getStyleSheet a utility function to generate a style sheet for a given text content
+ * @property {String} options.shadowId an id if undefined denotes that given root is a shadowRoot
+ * @property {Number} options.priority css applied priority
+ * @returns resolve with stylesheet object
  * @private
  */
-function loadCssom({ root, shadowId }, timeout, convertTextToStylesheetFn) {
-	/**
-	 * Make an axios get request to fetch a given resource and resolve
-	 * @method getExternalStylesheet
-	 * @private
-	 * @param {Object} param an object with properties to configure the external XHR
-	 * @property {Object} param.resolve resolve callback on queue
-	 * @property {Object} param.reject reject callback on queue
-	 * @property {String} param.url string representing the url of the resource to load
-	 * @property {Number} param.timeout timeout to about network call
-	 */
-	function getExternalStylesheet({ resolve, reject, url }) {
-		axe.imports
-			.axios({
-				method: 'get',
-				url,
-				timeout
-			})
-			.then(({ data }) => {
-				const sheet = convertTextToStylesheetFn({
-					data,
-					isExternal: true,
-					shadowId,
-					root
-				});
-				resolve(sheet);
-			})
-			.catch(reject);
-	}
-
-	const q = axe.utils.queue();
+function getExternalStylesheet(options) {
+	const {
+		resolve,
+		reject,
+		url,
+		rootNode,
+		timeout,
+		getStyleSheet,
+		shadowId,
+		priority
+	} = options;
+	axe.imports
+		.axios({
+			method: 'get',
+			url,
+			timeout
+		})
+		.then(({ data }) => {
```

Comment:
I don't recall off the top of my head. Does Axios treat non-2xx response codes as errors? If not, we might want to add something like:

```js
.then(({ data, status }) => {
  if ((status / 100 | 0) !== 2) {
    throw new Error('non-2xx response')
  }
  // ...
})
```

Reply:
Docs say that it does.